app/models/manageiq/providers/nuage/manager_mixin.rb

```diff
@@ -17,7 +17,7 @@ def raw_connect(username, password, endpoint_opts)
     end
 
     def auth_url(protocol, server, port, version)
-      scheme = protocol == "ssl-with-validation" ? "https" : "http"
+      scheme = (!protocol || protocol == "non-ssl") ? "http" : "https"
```

Comment:
@gasper-vrhovsek 
This is difficult to read. Can you describe the logic here so I can make sure I understand it correctly?

Reply:
Hi @bronaghs, sure i can describe:

If protocol is not provided or if 'non-ssl' is selected, then 'http' value is assigned. Otherwise 'https' is assigned.

My background is mostly java, so I am still getting used to Ruby. I'll search for a more readable solution.

Reply:
@bronaghs and @miha-plesko I have refactored the scheme selecting logic. Hope it's better and more readable now.